File: backend/app/schemas/medication.py
from pydantic import BaseModel


class MedicationOut(BaseModel):
    id: str
    name: str
    dose: str | None = None
    schedule: str | None = None
    times: list[str] = []
    is_anticoagulant: bool = False
    purpose: str | None = None
    taken_today: bool = False

    @classmethod
    def from_row(cls, m, taken_today: bool = False) -> "MedicationOut":
        return cls(
            id=str(m.id),
            name=m.name,
            dose=m.dose,
            schedule=m.schedule,
            times=[t.strip() for t in (m.times or "").split(",") if t.strip()],
            is_anticoagulant=m.is_anticoagulant,
            purpose=m.purpose_de,
            taken_today=taken_today,
        )

File: frontend_flutter/lib/models/medication.dart
class Medication {
  final String id;
  final String name;
  final String dose;
  final String schedule;
  final List<String> times;
  final bool isAnticoagulant;
  final String? purpose;
  final bool takenToday;

  const Medication({
    required this.id,
    required this.name,
    required this.dose,
    required this.schedule,
    required this.times,
    this.isAnticoagulant = false,
    this.purpose,
    this.takenToday = false,
  });

  factory Medication.fromJson(Map<String, dynamic> j) => Medication(
        id: j['id'].toString(),
        name: j['name'] as String,
        dose: (j['dose'] as String?) ?? '',
        schedule: (j['schedule'] as String?) ?? '',
        times: (j['times'] as List?)?.map((e) => e.toString()).toList() ?? const [],
        isAnticoagulant: (j['is_anticoagulant'] as bool?) ?? false,
        purpose: j['purpose'] as String?,
        takenToday: (j['taken_today'] as bool?) ?? false,
      );

  Medication copyWith({bool? takenToday}) => Medication(
        id: id, name: name, dose: dose, schedule: schedule, times: times,
        isAnticoagulant: isAnticoagulant, purpose: purpose, takenToday: takenToday ?? this.takenToday,
      );
}
